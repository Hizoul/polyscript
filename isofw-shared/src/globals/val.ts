const val = {
  isDebug: true,
  version: "0.1.0",
  i18nNameSpace: "langSpace",
  channel: {
    realtime: "realtime"
  },
  service: {
    user: "/users",
    project: "projects",
    camera: "cameras",
    preset: "presets",
    presetAssistant: "presetAssistant"
  },
  maximumPresetAmount: 100
}

export default val
