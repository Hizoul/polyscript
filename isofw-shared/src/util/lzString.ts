import * as lzString from "lz-string"

export default lzString
