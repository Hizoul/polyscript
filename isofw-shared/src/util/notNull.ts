const notNull: (value: any) => boolean = (value) => {
  return value !== undefined && value !== null
}

export default notNull
