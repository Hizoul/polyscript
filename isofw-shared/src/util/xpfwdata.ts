import {
  addId, AuthForm, BackendClient, dataOptions, DbStore, displayModeChanger,
  getListFormFromRelationshipField, IBackendClient, ICreateHookProps, IEditHookProps,
  IGetHookProps, IListHookProps, IRelationshipHookProps, IRemoveHookProps, IUiClient,
  ListStore, MailField, OwnerField, PwField, REMOVE_ADDON_KEY, removeId, searchRelated,
  useAuth, useCreate, useCreateWithProps, useEdit, useEditWithProps, useGet, useGetWithProps, toJS,
  useList, useListWithProps, useRelationship, useRelationshipWithProps, useRemove, useRemoveWithProps, UserStore
} from "@xpfw/data"

export {
  addId, AuthForm, BackendClient, dataOptions, DbStore, displayModeChanger,
  getListFormFromRelationshipField, IBackendClient, ICreateHookProps, IEditHookProps,
  IGetHookProps, IListHookProps, IRelationshipHookProps, IRemoveHookProps, IUiClient,
  ListStore, MailField, OwnerField, PwField, REMOVE_ADDON_KEY, removeId, searchRelated,
  useAuth, useCreate, useCreateWithProps, useEdit, useEditWithProps, useGet, useGetWithProps, toJS,
  useList, useListWithProps, useRelationship, useRelationshipWithProps, useRemove, useRemoveWithProps, UserStore
}
