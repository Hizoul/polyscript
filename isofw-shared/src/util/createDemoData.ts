import { BackendClient, DbStore } from "@xpfw/data"
import { FormStore, memo } from "@xpfw/form"
import val from "isofw-shared/src/globals/val"
import { makeRandomProgram } from "isofw-shared/src/network/causeProjectTraffic"
import { testCameras } from "isofw-shared/src/testUtil/data/camera"
import { addMappings, testProjects } from "isofw-shared/src/testUtil/data/project"
import { testUsers } from "isofw-shared/src/testUtil/data/users"

const demoLoadingKey = "demoData"
const demoTotalKey = "demoDataTotal"

const toCreate = [
  {collection: val.service.user, data: testUsers},
  {collection: val.service.camera, data: testCameras},
  {collection: val.service.project, data: testProjects}
]
let totalLength = 0
for (const create of toCreate) {
  totalLength += create.data.length
}
FormStore.setValue(demoTotalKey, totalLength)
addMappings()
const createDemoData = (onlyRemove: boolean) => {
  return async () => {
    FormStore.setLoading(demoLoadingKey, true)
    FormStore.setValue(demoLoadingKey, 0)
    let i = 0
    for (const create of toCreate) {
      for (const entry of create.data) {
        try {
          const removeRes = await DbStore.remove(entry._id, create.collection)
          console.log("REMOVERES IS", removeRes)
        } catch (e) {
          console.log("ERROR removing DEMODATA ", create.collection, entry, e)
        }
        try {
          if (!onlyRemove) {
            if (create.collection === val.service.project) {
              entry.program = makeRandomProgram()
            }
            const createRes = await BackendClient.client.create(create.collection, entry)
            console.log("CREATERES IS", createRes)
          }
        } catch (e) {
          console.log("ERROR create DEMODATA ", create.collection, entry, e)
        }
        FormStore.setValue(demoLoadingKey, ++i)
      }
    }
    FormStore.setLoading(demoLoadingKey, false)
  }
}
const useDemoData = () => {
  return {
    progress: FormStore.getValue(demoLoadingKey, undefined, 0),
    loading: FormStore.getLoading(demoLoadingKey),
    createData: memo(() => createDemoData(false), [demoLoadingKey, false]),
    removeDemoData: memo(() => createDemoData(true), [demoLoadingKey, true]),
    total: FormStore.getValue(demoTotalKey, undefined, 0)
  }
}

export default createDemoData
export {
  demoLoadingKey, useDemoData
}
