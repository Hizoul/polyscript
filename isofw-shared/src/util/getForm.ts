
import val from "isofw-shared/src/globals/val"
import { CameraForm } from "isofw-shared/src/xpfwDefs/camera"
import { PresetForm } from "isofw-shared/src/xpfwDefs/preset"
import { ProjectForm } from "isofw-shared/src/xpfwDefs/project"
import UserForm from "isofw-shared/src/xpfwDefs/user"

const getForm = (collection: string) => {
  switch (collection) {
    case val.service.camera: return CameraForm
    case val.service.preset: return PresetForm
    case val.service.project: return ProjectForm
    case val.service.user: return UserForm
    default: return undefined
  }
}

export default getForm
