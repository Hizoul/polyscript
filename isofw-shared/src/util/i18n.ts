import * as i18next from "i18next"
import * as sprintf from "i18next-sprintf-postprocessor"
import val from "isofw-shared/src/globals/val"

const i18nA: any = i18next
const toInitWith = i18nA
const i18n = toInitWith.init({
  lng: "en",
  ns: [val.i18nNameSpace],
  overloadTranslationOptionHandler: sprintf.overloadTranslationOptionHandler,
  defaultNS: val.i18nNameSpace,
  initImmediate: true,
  resources: {
    en: {
      main: {
        home: "Home",
        loggedInNavigate: "Already logged in. Do you want to navigate somewhere?",
        loggedInLogout: "Or do you wish to log out?",
        askLogin: "Please log in"
      }
    }
  }
})

export default i18n
