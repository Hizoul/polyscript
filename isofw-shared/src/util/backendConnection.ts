import { FeathersClient } from "@xpfw/data-feathers"
import url from "isofw-shared/src/globals/url"
import val from "isofw-shared/src/globals/val"
import makeBenchmarkClient from "isofw-shared/src/network/clientBenchmarker"
import { BackendClient, DbStore, UserStore } from "isofw-shared/src/util/xpfwdata"
import collections from "isofw-shared/src/xpfwDefs/collections"
import { ProjectForm } from "isofw-shared/src/xpfwDefs/project"

const connect = async (storage: any, getClient?: any) => {
  let clientToUse = getClient ? getClient() : FeathersClient
  if (clientToUse == null) {
    clientToUse = FeathersClient
  }
  BackendClient.client = clientToUse
  if (val.network.benchmarkEnabled) {
    BackendClient.client = makeBenchmarkClient(clientToUse, val.network.networkToUse)
  }
  // feathersClientOptions.batchService = val.service.batch
  const connectionOptions: any = {
    authOptions: {storage, timeout: 40000},
    port: url.port,
    makeAuth: true,
    useRest: false,
    userStore: UserStore,
    dbStore: DbStore,
    collections
}
  await BackendClient.client.connectTo(
    val.network.networkToUse === val.network.websocket ?
    `${url.webPrefix}${url.mainServer}:${url.port}` : url.mainServer, connectionOptions)

  for (const collection of collections) {
    BackendClient.client.client.service(collection).timeout = 40000
  }

  DbStore.formsToUpdate.push({
    collection: val.service.project,
    mapTo: String(ProjectForm.title),
    prefix: "edit"
  })
}

export default connect
