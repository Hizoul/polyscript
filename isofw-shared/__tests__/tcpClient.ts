import console = require("console");
import TCPClient from "isofw-node/src/network/client"
import getNetworkTestApp from "isofw-node/src/testUtil/getNetworkTestApp"
import val from "isofw-shared/src/globals/val"
import createTestUsers from "isofw-shared/src/testUtil/data/users"
import logIntoUser from "isofw-shared/src/testUtil/login"
import { BackendClient, DbStore, toJS, UserStore } from "isofw-shared/src/util/xpfwdata"

BackendClient.client = TCPClient
describe("tcp client test", () => {
  it("should work", async () => {
    console.log("PREPARING TCP SERVER")
    const tcpServer = await getNetworkTestApp(val.network.tcp, TCPClient)
    console.log("PREPARING TCP USER")
    const userResults = await createTestUsers(tcpServer.app)
    expect(toJS(UserStore)).toMatchSnapshot("Before Login")
    console.log("PREPARING TCP LOGIn")
    await logIntoUser()
    expect(toJS(UserStore)).toMatchSnapshot("After Login")
    await tcpServer.cleanUp()
  }, 8000)
})
