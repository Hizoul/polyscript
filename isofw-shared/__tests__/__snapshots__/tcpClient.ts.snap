// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tcp client test should work: After Login 1`] = `
Object {
  "authErr": Object {},
  "connected": true,
  "loading": false,
  "loggedIn": false,
  "user": null,
}
`;

exports[`tcp client test should work: Before Login 1`] = `
Object {
  "authErr": null,
  "connected": true,
  "loading": false,
  "loggedIn": false,
  "user": null,
}
`;
